rpc/backend/evm_backend.go

```diff
@@ -188,39 +156,15 @@ func (b *Backend) EthBlockFromTm(block *tmtypes.Block) (*ethtypes.Block, error)
 
 // GetTendermintBlockByNumber returns a Tendermint format block by block number
 func (b *Backend) GetTendermintBlockByNumber(blockNum types.BlockNumber) (*tmrpctypes.ResultBlock, error) {
-	height := blockNum.Int64()
-	currentBlockNumber, _ := b.BlockNumber()
-
-	switch blockNum {
-	case types.EthLatestBlockNumber:
-		if currentBlockNumber > 0 {
-			height = int64(currentBlockNumber)
-		}
-	case types.EthPendingBlockNumber:
-		if currentBlockNumber > 0 {
-			height = int64(currentBlockNumber)
-		}
-	case types.EthEarliestBlockNumber:
-		height = 1
-	default:
-		if blockNum < 0 {
-			return nil, errors.Errorf("cannot fetch a negative block height: %d", height)
-		}
-		if height > int64(currentBlockNumber) {
-			return nil, nil
-		}
-	}
-
-	resBlock, err := b.clientCtx.Client.Block(b.ctx, &height)
+	height := blockNum.TmHeight()
+	resBlock, err := b.clientCtx.Client.Block(b.ctx, height)
```

Comment:
I think there's an issue here, we do need to do a abci query to get the current block height in app state rather than the tendermint's block store.